torchgeo/trainers/detection.py

```diff
@@ -204,6 +207,9 @@ def configure_models(self) -> None:
         else:
             raise ValueError(f"Model type '{model}' is not valid.")
 
+        weight = adapt_input_conv(in_channels, self.model.backbone.body.conv1.weight)
+        self.model.backbone.body.conv1.weight = Parameter(weight)
```

Comment:
We should also modify the `in_channels` attribute of `model.backbone.body.conv1` because I just did this manually and even if you modify the weight it will stay `in_channels=3`. This won't show up in tests but could a silent bug later. 

```suggestion
        self.model.backbone.body.conv1.weight = Parameter(weight)
        self.model.backbone.body.conv1.in_channels = in_channels
```

Reply:
Just made this same change for instance segmentation too, thanks for catching this